<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <!-- 引入css -->
    <!-- <link rel="stylesheet" href="css/style.css" /> -->
    <link rel="stylesheet" href="css/engineering.css" />
    <link rel="stylesheet" href="icon/menu/iconfont.css">
    <script src="js/jquery.js"></script>
    <title>Engineering</title>
</head>
<style>
    @font-face {
        font-family: "Moon";
        src: url("font/MoonTypeFreeDemo-gwjr5.otf");
    }

    @font-face {
        font-family: "Milane";
        src: url("font/MilanelloFree-51Lwv.otf");
    }

    @font-face {
        font-family: "Tirto";
        src: url("font/TirtoWritterRegular-Eajrl.ttf");
    }

    @font-face {
        font-family: 'iconfont';
        src: url('iconfont.ttf?t=1708579574327') format('truetype');
    }
</style>

<body>
    <div class="BOX">
        <div class="TOP">
            <div>
              <p class="Level_1_heading">Project</p>
              <a href="safety.html" class="Secondary_headings">Safety</a>
              <a href="design.html" class="Secondary_headings">Design</a>
              <a href="implementation.html" class="Secondary_headings">Implementation</a>
          </div>
            <div>
              <p class="Level_1_heading">Wet Lab</p>
              <a href="protocol.html" class="Secondary_headings">Protocol</a>
              <a href="engineering.html" class="Secondary_headings">Engineering</a>
              <a href="Parts.html" class="Secondary_headings">Parts</a>
            </div>
            <div>
              <p class="Level_1_heading">Dry Lab</p>
              <a href="hardware.html" class="Secondary_headings">Hardware</a>
              <a href="software.html" class="Secondary_headings">Software</a>
              <a href="Model.html" class="Secondary_headings">Model</a>
            </div>
            <div>
              <p class="Level_1_heading">HP</p>
              <a href="intergrated HP.html" class="Secondary_headings">Intergrated HP</a>
              <a href="education.html" class="Secondary_headings">Education</a>
              <a href="entrepreneurship.html" class="Secondary_headings">Entrepreneurship</a>
              <a href="overview.html" class="Secondary_headings">Overview</a>
          </div>
            <div>
              <p class="Level_1_heading">Team</p>
              <a href="member.html" class="Secondary_headings">Member</a>
              <a href="attributions.html" class="Secondary_headings">Attributions</a>
            </div>
          </div>
        <div class="top">
            <img style="margin: 10px;margin-left: 5vw;" src="img/description/icon.png" id="iconclick">
            <div class="caidan"></div>
        </div>
        <div class="banner">
            <h1 class="title">Engineering</h1>
        </div>
        <div class="box">
            <div class="menu">
                <div class="blank"></div>
                <div class="nav">
                    <a href="#tit_1" class="Secondary_headings scroll">Module1:Efficacy Verification of AMP</a>
                    <a href="#tit_2" class="Level_3_heading scroll">Iteration1: Chemosynthetic AMP Test</a>
                    <a href="#tit_6" class="Level_3_heading scroll">Iteration2: Biosynthetic AMP Test</a>
                    <a href="#tit_11" class="Secondary_headings scroll">Module2:Sensor Switch and Control System</a>
                    <a href="#tit_12" class="Level_3_heading scroll">Iteration1: Separate Components Test</a>
                    <a href="#tit_17" class="Level_3_heading scroll">Iteration2: Optimisation of Component
                        Performance</a>
                    <a href="#tit_22" class="Level_3_heading scroll">Iteration3: Gene Circuit Assembly</a>
                    <a href="#tit_27" class="Level_3_heading scroll">Iteration4: Simulation Experiments Combined with
                        Hardware</a>
                    <a href="#tit_33" class="Secondary_headings scroll">Module3: Biology Safety</a>
                    <a href="#tit_34" class="Level_3_heading scroll">Iteration1: Separate Components Test</a>
                    <a href="#tit_39" class="Level_3_heading scroll">Iteration2: Optimisation of Component
                        Performance</a>
                    <a href="#tit_45" class="Secondary_headings scroll">Module 4: Part about Bacterial Hydrogels</a>
                    <a href="#tit_46" class="Level_3_heading scroll">Iteration1</a><br>
                    <a href="#tit_51" class="Level_3_heading scroll">Iteration2</a><br>
                    <a href="#tit_56" class="Level_3_heading scroll">Iteration3</a>
                    <!-- <div class="animation"></div> -->
                </div>
            </div>
            <div class="text" style="position: relative;">
                <div class="cover" style="width: 100%; height: 70px; position: sticky; top: 0%;"></div>
                <div>
                    <h1 id="tit_1" class="page">Module1:Efficacy Verification of AMP</h1>
                    <p>
                        In order to achieve a highly effective and specific targeting of Staphylococcus aureus in the
                        intestine. We have designed an antimicrobial peptide (AMP) - HHHHHHLLGSLLKLLPKLLCASYFCRWWWLL,
                        which has the characteristics of small molecule, targeting and high efficiency in theory. In
                        order to verify its practical effect, we decided to carry out experimental exploration step by
                        step from shallow to deep from chemical synthesis to biosynthesis.
                    </p>
                    <p>
                        Note: (The pH of the following experimental medium is 8.3).
                    </p>
                </div>
                <div>
                    <h2 id="tit_2" class="page">Iteration1: Chemosynthetic AMP Test</h2>
                </div>
                <div>
                    <p id="tit_3" class="bold page">Design</p>
                    <span>
                        1.Because of the inclusion of modified structures such as targeting domains and His-Tag in the
                        design, we first need to determine whether the designed and synthesized AMP can exert efficient
                        killing efficacy
                    </span><br>
                    <span>
                        2.Because MRSA is not the only microflora in the gut, we need to ensure that our AMP can kill M
                        RSA in low and efficient quantitieswithout affecting other intestinal colonies too much.
                    </span><br>
                    <span>
                        3.Although digestive enzymes are not secreted in the large intestine, residual digestive enzymes
                        are still present in the intestinal fluid. Although we have theoretically demonstrated that
                        there is no digestive enzyme-related site on AMP, further experimental verification is still
                        needed.
                    </span>
                </div>
                <div>
                    <p id="tit_4" class="bold page">Build</p>
                    <span>
                        1. AMP was chemically synthesized, and an appropriate amount of MRSA was cultured under the
                        condition of 1um concentration of AMP and the same concentration of normal saline, and its
                        growth was observed, and electron microscopy observation and flow cytometry analysis were
                        performed.
                    </span><br>
                    <span>
                        2. AMP was chemically synthesized, and MRSA, BS, EC, and PA were co-cultured at a concentration
                        of 1um AMP and the same concentration of normal saline, respectively.
                    </span><br>
                    <span>
                        3. AMP was chemically synthesized, an appropriate amount of trypsin and intestinal peptidase
                        were added to the medium, and then an appropriate amount of MRSA was cultured at a concentration
                        of 1um AMP and the same concentration of normal saline
                    </span>
                </div>
                <div>
                    <p id="tit_5" class="bold page">Test</p>
                    <span>
                        The growth of MRSA and other strains after treatment was observed, and electron microscopy and
                        flow cytometry analysis were performed.
                    </span>
                </div>
                <div>
                    <h2 id="tit_6" class="page">Iteration2: Biosynthetic AMP Test</h2>
                </div>
                <div>
                    <p id="tit_7" class="bold page">Design</p>
                    <span>After verifying the efficacy of AMP itself, we will perform simulated experiments on
                        biosynthesis in combination with specific application environments and chassis organisms.
                    </span><br>
                    <span>
                        1.First, we need to verify that chassis organisms can successfully express AMP after
                        modification, and how much AMP can be secreted. Because the amount of AMP secretion is directly
                        related to the effectiveness of our program and the size of side effects.
                    </span><br>
                    <span>
                        2.We need to verify that biosynthesized AMP has the same good biological effects as chemically
                        synthesized AMP
                    </span><br>
                    <span>
                        3.We are to see if the digestive enzymes and PH in the gut interact with chassis organisms that
                        cause other effects on the object.
                    </span>
                </div>
                <div>
                    <p id="tit_8" class="bold page">Build</p>
                    <span>
                        1. After the constructed plasmid is introduced into the chassis organism, it is cultured in
                        vitro for a period of time before being purified by immobilized metal ion affinity
                        chromatography (IMAC).
                    </span><br>
                    <span>
                        2. After the constructed plasmid was introduced into our chassis, the engineered chassis
                        organisms and the non-engineered chassis organisms were co-cultured with MRSA, respectively
                    </span><br>
                    <span>
                        3. After the constructed plasmid was introduced into our chassis, an appropriate amount of
                        trypsin and enteropeptidase was added to the culture medium, and then the modified chassis
                        organisms and the non-modified chassis organisms were co-cultured with MRSA
                    </span><br>
                    <span>
                        Test:
                        Fourth, verify whether the modified chassis organisms can successfully secrete AMP and determine
                        its secretion concentration
                    </span><br>
                    <span>
                        1. Purification by immobilized metal ion affinity chromatography (IMAC) was performed, and
                        purification and concentration determination of AMP were performed with the help of His-Tag
                    </span><br>
                    <span>
                        2.Its growth was observed, and electron microscopy and flow cytometry analysis were performed.
                    </span>
                </div>
                <div>
                    <p id="tit_9" class="page bold">Learn</p>
                    <span>
                        1.If the protein is not expressed (because it is too short, etc.), fluorescence in situ
                        hybridization can be used to see if the corresponding mRNA is expressed and whether the
                        corresponding gene is duplicated
                    </span><br>
                    <span>
                        2.If the protein is expressed but not potent, the pH of the medium can be changed for gradient
                        testing, and if the AMP is inactive due to pH, further residue substitution can be performed for
                        antimicrobial peptides
                    </span><br>
                    <span>
                        3.Protein expression, but no potency, if not pH, may be a structural problem with the chassis
                        biological modification (e.g., lack of disulfide bonds) that can be sequenced for purified
                        proteins
                    </span><br>
                    <span>
                        4.If the biosynthetic concentration is higher than required, the addition of silicons may be
                        considered.
                    </span><br>
                    <span>
                        5.If the synthetic concentration is lower than desired, repeats or an increase in the number of
                        imported plasmids may be considered
                    </span><br>
                    <span>
                        In order to verify whether the antimicrobial peptide we designed can meet the application needs
                        of our project, we designed a test to validate it
                    </span>
                </div>
                <div>
                    <p id="tit_10" class="bold page">References</p>
                    <ol>
                        <li>
                            L. Shang et al., “Hybrid Antimicrobial Peptide Targeting Staphylococcus aureus and
                            Displaying Anti-infective Activity in a Murine Model,” Front. Microbiol., vol. 11, p. 1767,
                            Sep. 2020, doi: 10.3389/fmicb.2020.01767.
                        </li>
                        <li>
                            B. Mishra and G. Wang, “Ab Initio Design of Potent Anti-MRSA Peptides Based on Database
                            Filtering Technology,” J. Am. Chem. Soc., vol. 134, no. 30, pp. 12426–12429, Aug. 2012, doi:
                            10.1021/ja305644e.
                        </li>
                    </ol>
                </div>
                <div>
                    <br>
                    <h1 id="tit_11" class="page">Module2:Sensor Switch and Control System</h1>
                    <p>
                        In order to better achieve the goal of targeted anti-Staphylococcus aureus therapy and reduce
                        antibiotic contamination, we believe that it is necessary to design an efficient, accurate,
                        sensitive and safe control system. Therefore, we designed a highly specific Staphylococcus
                        aureus sensor switch and a feedback mechanism to properly control AMP expression, which needs to
                        have positive feedback response and negative feedback modulation.
                    </p>
                </div>
                <div>
                    <h2 id="tit_12" class="page">Iteration1: Separate Components Test</h2>
                </div>
                <div>
                    <p id="tit_13" class="page bold">Design</p>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            For this purpose, we try to introduce Staphylococcus aureus Agr Quorum Sensing System,
                            including:The functional module of the Agr structure composed of agrC and agrA expressed
                            under
                            the T7 promoter is used for AIP signal sensing and intracellular transmission.with the P3
                            promoter, which acts as a switch for the AMP synthesis control module.[1], [2] [3]
                        </span><br>
                        <span>
                            But take into account AgrC protein. As membrane surface receptors require accurate membrane
                            surface representation, we try to construct E. coli surfaceslocalization of the fusion
                            protein
                            of peptide and AgrC,and carry out experimental tests on the surface display of the membrane,
                            including:lnK-N-his tag-AgrC，his tag-AgrCTwo experimental groups。[4]
                        </span><br>
                        <span>
                            In order to test the performance of the sensing element, we tried the design of the sfGFP
                            fluorophore attached to the downstream of the P3 promoter and constructed the relevant
                            strains,
                            and experimentally plotted the response curve of the fluorescent reporter gene to the
                            concentration and time, so as to explore the detection domain.
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback mechanism
                    </span><br>
                    <div>
                        <span>
                            In the beginning, we tried to useThe cre-loxp system builds a bidirectional switch,By
                            designing
                            the specific activation of the reverse loxp site and the cleavase enzyme, the regulatory
                            promoter of the E. coli transporter was shielded to create the membrane stress stimulated
                            state
                            inhibition AMP continues to be expressed. However, with the in-depth literature
                            investigation,
                            we believe that the system is generally similar to the design of disposable switches.It
                            doesn't
                            exactly match our highly reversible needs.[5]
                        </span><br>
                        <span>
                            Further investigation showed that lambda phages stably entered the lyogenic state through a
                            series of gene interactions and self-overexpression inhibition of Ci protein after
                            infection.
                            This gave us the inspiration to reinvent the feedback loop.
                        </span><br>
                        <p>
                            Cro will express and inhibit the PRM promoter with zero input, and after the input, we
                            designed
                            a fast-response expression pathway with positive feedback characteristics by expressing the
                            Ci
                            protein at the same time, and then through the inhibition of the Ci protein on the promoter
                            PR
                            expressing the cro, and the tandem structure of two reversed-phase devices to relieve its
                            own
                            transcriptional repression, and designed two experiments for simple verification
                        </p>
                        <span>
                            1.We found that there were reports in the literature that the PRM promoter was weak, and we
                            transferred a simple PRM-sfGFP sequence to explore the promoter efficiency.
                        </span><br>
                        <p>
                            2.Further, we designed an iPTG-induced sfGFP-containing plasmid to verify the design
                            efficiency
                            of our positive feedback pathway.
                        </p>
                        <span>
                            There are six Ci protein binding sites, OR1, OR2, OL1, OL2, OR3, and OL3, upstream and
                            downstream of the Ci protein sequence, and the Ci protein will prevent the initiation of
                            transcription here, and OR3 is located on the promoter PRM, Ci does not bind at low
                            concentrations,At high concentrations, it will form a DNA circular structure to promote the
                            formation of self-concentration clamping.[6]Based on this, we designed a negative feedback
                            regulatory pathway (PL-Ci) that clamps AMP expression levels-day-PRM-PR sequences were used
                            for
                            the experimental group) and in a system without ORs, OLs lociAs a control, the clamping
                            concentration and system stability were verified.
                        </span>
                    </div>
                    <div class="center">
                        <img src="img/experiment/图片10.png">
                        <div class="caption">Figure1 A. Plasmid functional sequence of iptg-induced positive feedback
                            pathway, with sfGFP
                            as the reporter gene;B. P RM-sfGFP sequence to explore promoter efficiency;The above figure
                            shows the design of a negative feedback regulatory pathway containing ORs and OLs, and the
                            control plasmid without feedback function is shown below. Plasmid functional sequences were
                            validated for AIP response, with sfGFP as the reporter gene; Different fusion proteins for
                            membrane display</div>
                    </div>
                </div>
                <div>
                    <span id="tit_14" class="page bold">Build</span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <p>
                            First, we will construct and electroporate the transduction plasmid into E. coli to
                            construct
                            the desired strain, and for the two experimental purposes mentioned above:
                        </p>
                        <span>
                            1.We collect and isolate cells, extract intracellular proteins of different membrane
                            components,
                            prepare samples for western blot analysis, and obtain membrane localization information of
                            the
                            target protein.[4]
                        </span><br>
                        <span>
                            2.We first cultured in normal medium without AIP to observe the fluorescent protein signal
                            intensity and assess the level of gene leakage, and then introduced a concentration gradient
                            of
                            synthetic AIP and Staphylococcus aureus cell lysates into the culture early in the bacterial
                            culture index and co-cultured for a period of time.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Feedback mechanism
                    </span><br>
                    <div>
                        <span>
                            We used a similar approach to construct the corresponding strains and express them.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_15" class="page bold">Test</span><br>
                    <span class="bold">
                        --“EcoTest”
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            In order to better model the transcription level, we have established a modeled "Eco Test"
                            technology route based on the existing conditions in the laboratory, which is of great help
                            to
                            us to determine the expression of proteins at all levels.
                        </span><br>
                        <span>
                            For the liquid bacterial samples to be measured, we first mix the culture medium and take a
                            certain sample, dilute it by a certain factor, measure the absorbance under OD600 of the
                            microplate reader to roughly determine the population density, and take samples at the same
                            time to obtain the gene transcription level by RT-qPCR.
                        </span><br>
                        <span>
                            Fluorescence intensity was determined at 488 nm excitation light for samples containing
                            reporter sfGFP, and ELISA method for samples containing his-tag was used to determine
                            concentration.
                        </span><br>
                        <span>
                            For a series of concentration gradients or time samples, we used the blank group as the
                            control to calculate the relative fluorescence intensity and plot the protein response
                            expression curve at a certain density, and calculated the relative transcription level and
                            plotted the transcription response curve according to the RT-qPCR results
                        </span><br>
                        <a href="protocol.html"> For specific experimental methods, click the link
                            to go to the protocol</a>
                    </div>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            For the outer protein display experiment, we repeated the experiment and analyzed the blot
                            results with image J, and obtained the gene sequence combination with the best localization
                            effect of the outer membrane display
                        </span><br>
                        <span>
                            For the response experiment, we calculated the gene leakage expression rate under the 0
                            concentration A IP group, and furthermore, we used the 0 concentration AIP group as the
                            control, and obtained the concentration response curve with the idea of "EcoTest" to verify
                            the sensitivity and efficiency of the sensor switch
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback mechanism
                    </span><br>
                    <div>
                        <span>
                            For PRM promoter validation experiments, PRM promoter efficiency was comprehensively judged
                            by "EcoTest" after the culture had entered the plateau period, which we thought would meet
                            our requirements, considering the low inhibitory concentration we needed.
                        </span><br>
                        <span>
                            For the validation experiment of the positive feedback pathway, we took the plasmid lacking
                            PR promoter and CRO coding as the control, and after entering the plateau period, iPTG was
                            added to the medium to induce the expression of his tag-CI, and then a batch of samples was
                            taken every 10 minutes.
                        </span><br>
                        <span>
                            For the validation experiment of the negative feedback pathway, a batch of samples was taken
                            every 30 minutes after several hours of shaker expansion, and the response curve was
                            obtained by "EcoTest", and it was observed that the experimental group Ci entered the steady
                            state of lower concentration earlier.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_16" class="page bold">Learn</span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We preliminarily verified the correct expression of Agr A and Agr C in E. coli, and membrane
                            localization experiments revealed the most efficient structure, i.e., Agr A, which is guided
                            by
                            the N-terminal domain of cold nuclear proteins, laying the structural foundation for the
                            correct
                            function of the sensing switch
                        </span><br>
                        <span>
                            We have made a preliminary exploration of the detection domain and response curve, and found
                            that the linear/nonlinear response of the following concentration range can be achieved, and
                            the
                            corresponding modeling will be carried out on this basis.
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback system
                    </span><br>
                    <div>
                        <span>
                            In the design process, we have a better understanding of some common methods of feedback
                            pathway
                            design in synthetic biology, and we also have a better understanding of the life cycle of
                            bacteriophages. The initial validation of the elements of the feedback pathway is helpful to
                            model and describe the AMP expression level and the amount of drug administration at that
                            time,
                            but at the same time, the current controllability of the system is not high enough, and we
                            hope
                            to find a genetic modification method that quantitatively controls the feedback depth of the
                            value. At the same time, more systematic issues such as the cascading effect of different
                            components and the burden on engineered bacteria need to be elaborated.
                        </span>
                    </div>
                </div>
                <div>
                    <h2 id="tit_17" class="page">Iteration2:Optimisation of component performance</h2>
                </div>
                <div>
                    <span id="tit_18" class="page bold">Design</span><br>
                    <span>
                        In the further study of engineering ideas, we realized that in the actual production
                        environment, the system will face many interferences in the complex environment, and we need to
                        consider the robustness of the system and verify and optimize it.
                    </span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            There are many different microbiota in the intestinal microenvironment, some of which are
                            probiotics, and we want to minimize the negative impact on these microbiota, so we designed
                            anti-crosstalk experiments and real-time experiments using the constructed strains.
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback system
                    </span><br>
                    <div>
                        <span>
                            Considering that our negative feedback pathway responds to environmental disturbances in
                            complex
                            environments, we hope to verify its dynamic response and steady-state output under Ci
                            disturbance through experiments.[7]
                        </span><br>
                        <span>
                            At the same time, through the analysis of the negative feedback mechanism, we guessed that
                            changing the length of the OR1-OL1 sequence may affect the final concentration level, and
                            experimentally verified this guess.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_19" class="page bold">Build</span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We designed the following experiments with reference to similar studies:[2]
                        </span><br>
                        <span>
                            Anti-crosstalk experiment: We obtained some common gram-positive pathogens in various
                            intestines
                            that grow, such as Staphylococcus aureus, Bacillus subtilis, etc., with Bifidobacterium as a
                            variable, called "sensing bacteria";Firstly, different types of "inductive bacteria" were
                            inoculated in a blank liquid medium, and after the appropriate time was cultivated, each
                            engineering bacteria were inoculated, and the medium without inductive bacteria was used as
                            a
                            control.
                        </span><br>
                        <span>
                            Real-time experiments: Literature investigation found that erythromycin and aminobutyric
                            acid
                            can change the ability of bacteria to secrete AIP, and we can add corresponding drugs in the
                            co-culture environment to induce changes in the concentration of A. aureus secretion,so as
                            to
                            observe whether our sensor switch can achieve timely response.
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            Perturbation experiment: We transferred the functional fragment with atc-induced expression
                            of
                            Ci into the original strain, and added an appropriate amount through the medium after the
                            culture entered the plateau stageatc, which disturbs Ci homeostasis.
                        </span><br>
                        <span>
                            Sequence adjustment experiment: Based on the original plasmid, we designed nonsense
                            nucleotide
                            sequences ranging from 30 bp to 180 bp downstream of Ci, repeated the relevant experiments
                            involved in the first round and verified them by WB.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_20" class="page bold">Test</span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            The "EcoTest" test was carried out with the non-engineered bacteria group as the control and
                            multiple replicates were performed, in which different sensing bacteria should be used as
                            different groups for the anti-crosstalk test, and for the real-time experiment, samples
                            should
                            be taken at a certain interval after adding the drug
                        </span><br>
                        <span>
                            Anti-crosstalk experiment: (assuming good effect) the fluorescence intensity of the other
                            strains was lower than that of the control group with no sensors, while the Staphylococcus
                            aureus group had a significantly stronger fluorescence signal
                        </span><br>
                        <span>
                            Real-time experiment: (assuming a good effect) verifies the accuracy of a round of detection
                            domain, in which the fluorescence intensity is sensitively up/down-regulated in a short
                            period
                            of time
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback system
                    </span><br>
                    <div>
                        <span>
                            Perturbation experiment: Samples are taken at intervals and detected and plotted with
                            "EcoTest".
                        </span><br>
                        <span>
                            Sequence adjustment experiment: The protein expression level and gene transcription level in
                            the
                            steady state were obtained by "EcoTest", and the blot results were analyzed by image J to
                            confirm the protein expression level.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_21" class="page bold">Learn</span><br>
                    <span class="bold">
                        --Sensor switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            1.Anti-Crosstalk Experiment:
                            The results indicated that the sensor switch had a good anti-crosstalk effect and good
                            targeting, and had the potential to induce Staphylococcus aureus in a complex intestinal
                            microenvironment
                        </span><br>
                        <p>
                            If the fluorescence intensity is not accurate, the experiment can be repeated by S DS-PAGE
                            quantification and flow cytometry to determine the fluorescence intensity; it may be the
                            interference of sensing bacteria, and then the bacteria + medium environmental extract can
                            be
                            used as the medium substrate by cultivating the corresponding bacteria, and even
                            synthesizing
                            the A IP sub of different strainstype to carry out the experiment
                        </p>
                        <p>
                            2.Real-time experiments:
                            It indicates that it has good real-time performance, good targeting, and has the potential
                            to
                            induce Staphylococcus aureus in a complex intestinal microenvironment.
                        </p>
                        <span>
                            2.1 round: It may be that the expression of Agr protein is not enough, and then the
                            experiment
                            can be repeated by increasing its expression; if the fluorescence intensity detection is not
                            accurate, the experiment can be repeated by S DS-PAGE quantification and flow cytometry to
                            measure the fluorescence intensity; it may be that the experimental quantification method is
                            not
                            accurate enough, the total protein can be extracted by rapid collection and then detected;
                            by
                            RBS, promoter orientation optimization, etc., to obtain sequences with higher expression
                            efficiency.
                        </span>
                    </div>
                    <span class="bold">
                        --Feedback system
                    </span><br>
                    <div>
                        <span>
                            We have verified the component performance of the feedback path, perhaps to supplement the
                            existing component data. In addition, we initially tried a simple method to control the
                            expression concentration of AMP, which will be more convenient for our assembly design, but
                            perhaps its performance will be unstable, and the leakage and expression of nonsense
                            nucleotide
                            sequences may also interfere with the normal function, so in addition, we can further try to
                            optimize the system by synthesizing promoters and adding enhancers.
                        </span>
                    </div>
                </div>
                <div>
                    <h2 id="tit_22" class="page">Iteration3: Gene Circuit Assembly</h2>
                </div>
                <div>
                    <span id="tit_23" class="page bold">Design</span><br>
                    <p>
                        After validating and optimizing each discrete functional element, we try to splice it into a
                        complete gene circuit to build a gene that can autonomously respond to and control exosome AMP
                        engineering bacteria are mainly divided into three steps:
                    </p>
                    <span>
                        ①PRE promoter efficiency verification: Cii, Ciii protein can activate the expression of PRE,
                        which was selected as the signaling molecule that connects the sensing switch to the feedback
                        pathway, and we first constructed the relevant vector and verified the PRE promoter efficiency.
                    </span><br>
                    <span>
                        ②Connection between the sensor switch and the feedback path: Connect the Cii and Ciii to the
                        downstream of the response system to test whether they can be connected.
                    </span><br>
                    <span>
                        ③Connect the control system to the AMP series: Connect the AMP to the upstream of Ci to test
                        whether the AMP is expressed normally
                    </span><br>
                    <div class="center">
                        <img src="img/experiment/图片11.png">
                        <div class="caption">
                            Figure 2 We connect the system in three steps, figure shows the sequence fragment of our new
                            design and its position in the final pathway, and the numbers correspond to the three steps
                            described in the text.
                        </div>
                    </div>
                </div>
                <div>
                    <span id="tit_24" class="page bold">Build</span><br>
                    <span>
                        We follow the idea of BioBricks, use the existing plasmid as the backbone to carry out modular
                        connection, step by step to construct the corresponding strains, and repeat the previous
                        experiments under stable culture conditions.
                    </span>
                </div>
                <div>
                    <span id="tit_25" class="page bold">Test</span><br>
                    <span>
                        We used the corresponding vector template plasmid as a blank control to determine the relevant
                        response curves to verify whether the overall response and feedback pathway were connected, and
                        whether AMP could be adjusted to the appropriate expression level.
                    </span>
                </div>
                <div>
                    <span id="tit_26" class="page bold">Learn</span><br>
                    <span>
                        We have successfully constructed an engineered bacteria that release AMP in a controlled manner,
                        and the next step may be to consider more realistic use scenarios, as well as possible problems
                        such as low efficiency and large chassis bioload, we may try to optimize by further directed
                        evolution, streamlining sequences, etc.
                    </span>
                </div>
                <div>
                    <h2 id="tit_27" class="page">Iteration4:Simulation Experiments Combined with Hardware</h2>
                </div>
                <div>
                    <span id="tit_28" class="page bold">Design</span><br>
                    <span>
                        Considering that the complex factors of application may have an unpredictable impact on the
                        achievement effect, we hope to build an integrated intestinal microarray testing system to test
                        the colonization ability of engineered bacteria, the ability to interact with each other in the
                        intestinal microbiota, and the killing ability of Staphylococcus aureus.
                    </span><br>
                    <span>
                        The test system includes an intestinal chip, an electronic measurement system and related
                        processing procedures, and the three modules are directly combined and assembled to build a
                        complete test system.
                    </span><br>
                    <span>
                        The specific design details are detailed in our "Hardware" section.
                    </span>
                </div>
                <div>
                    <span id="tit_29" class="page bold">Build</span><br>
                    <span>
                        Again, the specific construction details are described in more detail in our "Hardware" section.
                    </span>
                </div>
                <div>
                    <span id="tit_30" class="page bold">Test</span><br>
                    <span>
                        A Staphylococcus aureus infection model was constructed by introducing Staphylococcus aureus
                        strains on intestinal microarrays through co-culture.
                    </span><br>
                    <span>
                        IVIS imaging was performed at different time points using the fluorescent protein labeling of
                        the engineered bacteria, which showed the distribution and density of E. coli at different
                        stages of infection, and the response of the engineered bacteria to Staphylococcus aureus could
                        be observed.
                        At the same time, at different time points, the latex agglutination test was carried out by
                        taking the outer base culture medium, and the corresponding distribution density of
                        Staphylococcus aureus was obtained, reflecting the killing effect of Escherichia coli on
                        Staphylococcus aureus at the corresponding time points.
                    </span>
                </div>
                <div>
                    <span id="tit_31" class="page bold">Learn</span><br>
                    <span>
                        The density and distribution of engineered bacteria obtained by hardware can reflect the
                        response effect of E. coli in practical applications, indicating that our design can be verified
                        at the response level of engineered bacteria.
                    </span><br>
                    <span>
                        The density of Staphylococcus aureus in the corresponding period can characterize the killing
                        efficiency of Escherichia coli, and verify that the engineered bacteria we designed can have a
                        good killing effect on Staphylococcus aureus in the near in vivo environment, which has a
                        certain degree of reference significance for specific clinical applications.
                    </span>
                </div>
                <div>
                    <span id="tit_32" class="page bold">Reference</span><br>
                    <ol>
                        <li>
                            Di H., Lan L., and Chen F., “Small molecules targeting Staphylococcus aureus virulence,”
                            Chin. Sci. Bull., vol. 58, no. 36, pp. 3743–3752, Dec. 2013, doi: 10.1360/972013-351.
                        </li>
                        <li>
                            D. Lubkowicz, C. L. Ho, I. Y. Hwang, W. S. Yew, Y. S. Lee, and M. W. Chang, “Reprogramming
                            Probiotic Lactobacillus reuteri as a Biosensor for Staphylococcus aureus Derived AIP-I
                            Detection,” ACS Synth. Biol., vol. 7, no. 5, pp. 1229–1237, May 2018, doi:
                            10.1021/acssynbio.8b00063.
                        </li>
                        <li>
                            T. J. Polaske, K. H. J. West, K. Zhao, D. L. Widner, J. T. York, and H. E. Blackwell,
                            “Chemical and Biomolecular Insights into the Staphylococcus aureus Agr Quorum Sensing
                            System: Current Progress and Ongoing Challenges,” Israel Journal of Chemistry, vol. 63, no.
                            5–6, p. e202200096, Jun. 2023, doi: 10.1002/ijch.202200096.
                        </li>
                        <li>
                            S. Bao et al., “Construction of a cell-surface display system based on the N-terminal domain
                            of ice nucleation protein and its application in identification of Mycoplasma adhesion
                            proteins,” J Appl Microbiol, vol. 119, no. 1, pp. 236–244, Jul. 2015, doi:
                            10.1111/jam.12824.
                        </li>
                        <li>
                            J. J. Gierut, T. E. Jacks, and K. M. Haigis, “Strategies to Achieve Conditional Gene
                            Mutation in Mice,” Cold Spring Harb Protoc, vol. 2014, no. 4, p. pdb.top069807, Apr. 2014,
                            doi: 10.1101/pdb.top069807.
                        </li>
                        <li>
                            I. B. Dodd, A. J. Perkins, D. Tsemitsidis, and J. B. Egan, “Octamerization of λ CI repressor
                            is needed for effective repression of P RM and efficient switching from lysogeny,” Genes
                            Dev., vol. 15, no. 22, pp. 3013–3022, Nov. 2001, doers: 10.1101/gad.937301.
                        </li>
                        <li>
                            S. K. Aoki, G. Lillacci, A. Gupta, A. Baumschlager, D. Schweingruber, and M. Khammash, “A
                            universal biomolecular integral feedback controller for robust perfect adaptation,” Nature,
                            vol. 570, no. 7762, pp. 533–537, Jun. 2019, doi: 10.1038/s41586-019-1321-1.
                        </li>
                    </ol>
                </div>
                <div>
                    <br>
                    <h1 id="tit_33" class="page">Module3:Biology Safety</h1>
                    <p>
                        We have designed several layers of biosafety modules based on different levels of life for
                        different safety purposes, and for the above work, we have designed rigorous experiments to
                        explore the biocontainment efficiency of relevant modules.
                    </p>
                </div>
                <div>
                    <h2 id="tit_34" class="page">Iteration1: Separate Components Test</h2>
                </div>
                <div>
                    <span id="tit_35" class="page bold">Design</span><br>
                    <span class="bold">
                        -- DD-SHIELD1 system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We designed a system by which transcription arrest is artificially controlled by small
                            molecules
                            and design a series of experiments for verification, where we take sfGFP as the reporter .
                        </span><br>
                        <a href="design.html">The specific design can be found here.</a><br>
                    </div>
                    <span class="bold">
                        --Cold-sensitive suicide device
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We devised a genetic pathway for cryogenic signaling to activate E. coli suicide to reduce
                            the
                            risk of environmental contamination when excreted.
                        </span><br>
                        <a href="design.html">
                            The specific design can be found here.
                        </a>
                    </div>
                    <span class="bold">
                        --Quorum sensing
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We devised a suicide mechanism based on a quorum sensing system to control the density of E.
                            coli in the gut.
                        </span>
                        <a href="design.html">We devised a suicide mechanism based on a quorum sensing system to control
                            the
                            density of E. coli in the gut.</a>
                        <span>Firstly, We add sfGEP to the pathway as reporter genes, and then toxin protein gene was
                            added
                            in the upstream to verify the density control effect and stability under perturbation after
                            proof of concept.
                        </span>
                    </div>
                    <span class="bold">
                        --Horizontal gene transfer
                    </span><br>
                    <div>
                        <span>
                            In addition, horizontal gene transfer has always been an important cause of genetic
                            contamination caused by transgenic engineering, and we designed a batch of experiments to
                            test
                            the frequency of intraspecific and interspecific binding transfer of bacteria,
                            respectively，Make
                            sure that the environmental impact of the engineered bacteria is small enough.
                        </span><br>
                    </div>
                    <div class="center">
                        <img src="img/experiment/图片12.png">
                        <div class="caption">Figure3 A. Quorum sensing system pathway, B. Plasmid sequence for
                            dd-shield1 validation, C Toggle Switch sequences with different fluorescent reporter genes
                            at both ends</div>
                    </div>
                </div>
                <div>
                    <span id="tit_36" class="page bold">Build</span><br>
                    <span class="bold">
                        -- DD-SHIELD1 system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We cultured the corresponding strains and inoculated them in liquid medium with the addition
                            of
                            different concentrations Shield1 , react for a while.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Cold-sensitive suicide device
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We constructed a cold-sensitive promoter-toxin system with an empty plasmid/strong
                            promoter/empty medium as a control and incubated for a period of time at a temperature
                            gradient
                            of 5 °C, 10 °C, 15 °C, 20 °C, 25 °C, 30 °C, 35 °C, 40 °C, and further, repeated the
                            experiment
                            on a more detailed density gradient of 30-40.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Quorum sensing
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We construct the corresponding strains, apply LACL externally, and culture for a long enough
                            time.
                        </span><br>
                        <span>
                            On the basis of the above, the strains constructed with toxin proteins entered the plateau
                            stage, centrifuged, collected bacteria, and resuspended at twice the amount of bacteria or
                            twice
                            the amount of medium.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Horizontal gene transfer
                    </span><br>
                    <div>
                        <span>
                            We transferred intra-ECN carrying p-tetracycline, kanamycin, and
                            ammoniabenzicillin-resistant
                            plasmid RP4, Intraspecific transfer was validated by Pseudomonas putinosa, which carries the
                            same plasmid, and interspecific conjugate transfer was validated by Escherichia coli
                            NK5449-resistant strains. The above mixed culture was inoculated into LB plates of different
                            antibiotic groups after 16 h, and the gene transcription level was measured by the number of
                            viable colonies and RT-qPCR.[1]
                        </span><br>
                    </div>
                </div>
                <div>
                    <span id="tit_37" class="page bold">Test</span><br>
                    <span class="bold">
                        -- DD-SHIELD1 system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            After sample dilution, the microplate reader measured the fluorescence intensity at 488nm,
                            and
                            prepared samples for western blotting experiments to determine the protein expression.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Cold-sensitive suicide device
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            After sampling dilution, the microplate reader measured the microbial density at OD600 and
                            repeated to obtain the temperature-density curves of the three groups of cold sensitive
                            promoters/empty plasmids/strong promoters.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Quorum sensing
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            Samples were taken to measure the absorbance to obtain the colony growth curve and AHL
                            change
                            curve, as well as the colony concentration threshold represented by the expression of
                            fluorescent protein, and to observe whether the bacterial amount was maintained around this
                            threshold through the expression of toxin protein, and whether it could return to the
                            threshold
                            after changing the perturbation of bacterial density.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Horizontal gene transfer
                    </span><br>
                    <div>
                        <span>
                            The number of bacteria grown on the medium with four antibiotics (including tetracycline,
                            kanamycin, ampicillin, and rifampicin) divided by the number of bacteria grown on the medium
                            containing rifampicin-only, calculate the calculated horizontal gene transfer rate. Using
                            the
                            results of RT-qPCR,Use the formula to calculate the relative expression of each gene，16-S
                            rRNA
                            is quantified as a referenceto calculate the horizontal gene transfer rate
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_38" class="page bold">Learn</span><br>
                    <span>
                        Through multi-level biosafety transformation, we have built a good biocompatible engineering
                        chassis, but whether excessive modification will cause inefficiency of functional modules and
                        limited application scenarios. In the future, we will consider further designing safety systems
                        with more controllable performance.
                    </span>
                </div>
                <div>
                    <h2 id="tit_39" class="page">Iteration2:Optimisation of Component Performance</h2>
                </div>
                <div>
                    <span id="tit_40" class="page bold">Design</span><br>
                    <span class="bold">
                        --Quorum sensing system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            In the process of constructing the transduction plasmid, we found that the sequences
                            involved
                            were relatively large, which would make the transduction efficiency of E. coli inefficient,
                            so
                            we tried to make some improvements to this. So we removed tetO and antiholin sequence from
                            the
                            reverse switch, while it would limit the number of preventive measures to prevent gene
                            leakage,
                            which would require experimental validation.
                        </span><br>
                        <a href="design.html">The updated gene circuit design can be found here.</a><br>
                    </div>
                    <span class="bold">
                        --Luxbox
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            In order to achieve a controlled adjustment of quorum sensing density, Lux was
                            introduced-box
                            system, and discover,[2] luxbox sequence and The number of CRPBS changes the gene expression
                            on
                            both sides, thereby adjusting the population density. We constructed two batches of
                            plasmids,
                            one of which was from a different sourceluxbox sequence, a set of different numbers CRPBS
                            locus.
                        </span><br>
                        <a href="design.html">The specific design can be found here.</a><br>
                    </div>
                    <span class="bold">-- Toggle switch</span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            Although we have complete biosecurity measures, for the convenience of production, we wanted
                            to
                            design a bidirectional steady state switch to shield the suicide loop during the production
                            process. To test this concept, we designed the following sequences, with GFP and mCherry as
                            reporter genes reporting the switching state to us.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Assemble
                    </span><br>
                    <div>
                        <span>
                            We try to connect DD-SHIELD1 to the functional pathway to make it effective in practical
                            application scenarios.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_41" class="page bold">Build</span><br>
                    <span class="bold">
                        --Quorum sensing system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We used a newly constructed quorum sensing system to culture in blank medium to observe the
                            signal intensity of fluorescent proteins in the exponential early stage and assess the level
                            of
                            gene leakage.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Lux-box
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            The corresponding strains were constructed and cultured, and the above quorum sensing
                            experiments were repeated to observe the actual effect.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Toggle switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            First, we construct and electroporate the transduction plasmid into E. coli to construct the
                            corresponding strain, and after a period of propagation, next experiments were carried out
                            at
                            25°C. Changes in fluorescence intensity were observed in media containing ClpXP and aTc,
                            respectively.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Assemble
                    </span><br>
                    <div>
                        <span>
                            The corresponding strains were constructed and cultured, and repeated after the target
                            bacteria
                            entered the plateau phasePart of the experiment of the DD-SHIELD1 system.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_42" class="page bold">Test</span><br>
                    <span class="bold">
                        --Quorum sensing system
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            By measuring downstream fluorescent protein expression levels and sampling to determine
                            transcription levels, it was identified that gene leakage was at a low level with little
                            impact
                            on the normal growth of E. coli.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Lux-box
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            Through "EcoTest" The expression intensity curve was measured and the change in density
                            threshold in the presence of toxin proteins was determined.
                        </span><br>
                    </div>
                    <span class="bold">
                        -- Toggle switch
                    </span><br>
                    <div style="margin-bottom: 5px;">
                        <span>
                            We used a microplate reader to measure the fluorescence intensity, incubated at 42°C, the
                            inducer-free group was used as a control, the typical field of view was taken under the
                            microscope after coating culture, the relative fluorescence intensity was calculated and
                            compared with the RT-qPCR results to obtain the switching expression intensity and leakage
                            rate.
                        </span><br>
                    </div>
                    <span class="bold">
                        --Assemble
                    </span><br>
                    <div>
                        <span>
                            Through "EcoTest" The expression inhibition curve was determined to obtain a small molecule
                            drug
                            concentration with good utility.
                        </span>
                    </div>
                </div>
                <div>
                    <span id="tit_43" class="page bold">Learn</span><br>
                    <span>Luckily, the verification of toggle switch has been successful, which has a strong conversion
                        efficiency, a lower leakage rate, a faster conversion speed, and better meet the effect of
                        engineering bacteria production and taking. For its existing shortcomings, we will consider
                        replacing it with a higher efficiency promoter to improve the conversion efficiency, changing a
                        stronger terminator, to reduce gene leakage and other ways to improve.
                    </span><br>
                    <span>
                        Our new quorum sensing system tries to take into account the cellular burden and good functional
                        implementation, and can manually change the set population threshold, which broadens the use
                        scenarios of the system and facilitates the more refined regulation of colony density.
                    </span><br>
                </div>
                <div>
                    <span id="tit_44" class="page bold">Reference</span><br>
                    <ol>
                        <li>
                            J. Zhang et al., “Inhibition of quorum sensing serves as an effective strategy to mitigate
                            the risks of human bacterial pathogens in soil,” Journal of Hazardous Materials, vol. 465,
                            p. 133272, Mar. 2024, doi: 10.1016/j.jhazmat.2023.133272.
                        </li>
                        <li>
                            C. Ge et al., “Redesigning regulatory components of quorum-sensing system for diverse
                            metabolic control,” Common Nat, vol. 13, no. 1, p. 2182, Apr. 2022, doi:
                            10.1038/s41467-022-29933-x.
                        </li>
                    </ol>
                </div>
                <div>
                    <br>
                    <h1 id="tit_45" class="page">Module 4: Part about Bacterial Hydrogels</h1>
                    <p>
                        It consists of three test cycles for the construction of a hydrogel system that can be secreted
                        by E. coli.
                    </p>
                    <p>
                        Iteration1 : Measurement of hydrogel biosynthetic amount
                    </p>
                    <p>
                        Iteration2 : Tests of the physicochemical properties of hydrogels
                    </p>
                    <p>
                        Iteration3 : Engineered bacteria adhesion ability and safety test
                    </p>
                </div>
                <div>
                    <h2 id="tit_46" class="page">Iteration1</h2>
                </div>
                <div>
                    <span id="tit_47" class="page bold">Design</span><br>
                    <span>
                        Understanding the key role of the mucosal system outside the intestinal epithelial cells in
                        intestinal homeostasis, we hope to construct a hydrogel that can be secreted by microorganisms
                        to antagonize the colonization of pathogenic bacteria, repair the intestinal mucosal barrier,
                        and enhance the adhesion and colonization ability of engineered bacteria.
                    </span><br>
                    <span>
                        To this end, referring to previous studies, we designed a triplet chimeric protein, the
                        frizzling protein CsgA, the tag 6×His for ligation and examination, and the cytokine TFF, based
                        on the trilobite factor (TFF) of the fibrous matrix of the intestinal mucosa itself. It was
                        expected that TFF would be tethered to a crimped fiber matrix to construct a self-crimping
                        hydrogel.
                    </span><br>
                    <span>
                        Among them, CsgA itself is a common protein component in the extracellular matrix of bacteria,
                        and is a class of frizzled proteins that can be used as scaffolds for active biomaterials.
                    </span><br>
                    <span>
                        TFF, on the other hand, is a small protein secreted by mucus-producing cells on the surface of
                        the gastrointestinal tract and other mucous membranes, which can be used to promote epithelial
                        recovery and inhibit intestinal inflammation. According to the different TFFs, three chimeric
                        proteins, cTFF1, cTFF2 and cTFF3, were designed for subsequent production and testing.
                    </span>
                </div>
                <div>
                    <span id="tit_48" class="page bold">Build</span><br>
                    <span>
                        In order to achieve successful modification and secretion of chimeric proteins, they need to be
                        co-transcribed with the necessary modification genes (csgB, csgC, csgE, csgF, csgG). Therefore,
                        we co-composed their genes into a coiled protein expression operon that wasintroduced into the
                        kanamycin select-tagged pBbB8k plasmid backbone using P BAD as the promoter.
                    </span><br>
                    <span>
                        The plasmid was electrotransduced into E. coli to obtain the corresponding three types of
                        strains. The transformed bacteria are cultured at 37 °C in high osmolality medium to mimic
                        physiological conditions and induced with arabinose.
                    </span>
                </div>
                <div>
                    <span id="tit_49" class="page bold">Test</span><br>
                    <span>
                        Using the quantitative Congo red binding (CR) assay, the expression efficiency of different
                        fusion methods could be investigated to explore whether the three CsgA-TFF fusion proteins could
                        be expressed and assembled into crimped fibers under predetermined conditions under wild-type
                        control.
                    </span><br>
                    <span>
                        Using a whole-cell enzyme-linked immunosorbent assay (ELISA), the 6xHIS tag was probed to
                        further confirm the extracellular assembly.
                    </span>
                </div>
                <div>
                    <span id="tit_50" class="page bold">Learn</span><br>
                    <span>
                        Through this round of experiments, it was confirmed that the chimeric proteins we designed could
                        be expressed and assembled as crimped fibers under predetermined conditions, demonstrating the
                        usability of the hydrogel system.
                    </span><br>
                    <span>
                        In addition, the expression and assembly efficiency of chimeric proteins of different designs
                        were preliminarily shown through control experiments, which provided ideas for the selection and
                        optimization of final hydrogel proteins.
                    </span>
                </div>
                <div>
                    <h2 id="tit_51" class="page">Iteration2</h2>
                </div>
                <div>
                    <span id="tit_52" class="page bold">Design</span><br>
                    <span>
                        Through this round of experiments, it was confirmed that the chimeric proteins we designed could
                        be expressed and assembled as crimped fibers under predetermined conditions, demonstrating the
                        usability of the hydrogel system.
                    </span><br>
                </div>
                <div>
                    <span id="tit_53" class="page bold">Build</span><br>
                    <span>
                        Bacterial cultures are vacuum-filtered to obtain concentrated biomass. After treatment with 5%
                        SDS solution, wash well with water. The concentrated hydrogel is scraped on the membrane and
                        consists of 95–97% water. This hydrogel can be stored in a humid environment at 4°C or
                        lyophilized into powder form to meet our application requirements.
                    </span><br>
                </div>
                <div>
                    <span id="tit_54" class="page bold">Test</span><br>
                    <span>
                        Scanning electron microscopy (SEM) was used to characterize the hydrogels, which could reveal
                        the gelatinization effect and the network structure of the hydrogels at a microscopic level.
                        Rheological frequency sweeps can understand the response of a material to impact or gradual
                        loading. The storage modulus and loss modulus of the hydrogel can be obtained by rheological
                        scanning, and the mechanical properties of the gel such as inertia, impact resistance, and
                        damping properties can be understood.
                    </span><br>
                </div>
                <div>
                    <span id="tit_55" class="page bold">Learn</span><br>
                    <span>
                        The SEM results of the samples show the nanostructure of the hydrogel material, which can prove
                        the successful gel formation of the designed hydrogel, and at the same time, the excellent
                        biomimetic properties of the material can be verified by comparing it with the natural
                        structure.
                        The results of rheological analysis can show the excellent physical and chemical properties of
                        hydrogels, and the influence of different factors on the properties of the materials can be
                        probed, which can provide a reference for the adjustment under actual application.
                    </span><br>
                </div>
                <div>
                    <h2 id="tit_56" class="page">Iteration3</h2>
                </div>
                <div>
                    <span id="tit_57" class="page bold">Design</span><br>
                    <span>
                        Considering the need for long-lasting colonization of engineered bacteria in specific use cases,
                        it is necessary to evaluate the adhesion characteristics and safety of live bacteria hydrogels.
                        To this end, we tried to simulate different levels of the biological environment, combined with
                        the design of hardware, to study the different properties of hydrogels.
                    </span><br>
                </div>
                <div>
                    <span id="tit_58" class="page bold">Build</span><br>
                    <span>
                        Simulates the construction of the gut, and the specific details are elaborated in more detail in
                        the "Hardware" section.
                    </span><br>
                </div>
                <div>
                    <span id="tit_59" class="page bold">Test</span><br>
                    <span>
                        The adhesion characteristics of the gel, the residence of the gel and the colonization ability
                        of the engineered bacteria could be characterized by labeling the live gel with Cy7, staining
                        with Congo red, visualizing the gel, and imaging the simulated intestine with IVIS at different
                        time points.
                    </span><br>
                    <span>
                        Co-culturing engineered bacteria in an in vitro model of a human colorectal adenocarcinoma cell
                        line (Caco-2) can confirm whether overproduction of modified crimped fibers induces a pathogenic
                        phenotype of PBP8.
                    </span>
                </div>
                <div>
                    <span id="tit_60" class="page bold">Learn</span><br>
                    <span>
                        The results of gel adhesion experiments can prove the effectiveness of the introduction of
                        hydrogel system in enhancing the adhesion ability of engineered bacteria, indicating that
                        hydrogels and engineered bacteria can achieve long-term continuous colonization in the
                        intestine, which verifies the feasibility of the desired long-lasting colonization effect of the
                        intestinal tract to a certain extent.
                    </span><br>
                    <span>
                        The co-culture experiments of Caco-2 showed that the loaded engineering bacteria had low
                        translocation and low invasiveness, and the overproduction of frizzled protein did not induce
                        the pathological phenotype, which provided a certain degree of guarantee for the safety of
                        clinical administration.
                    </span>
                </div>
                <div>
                    <span id="tit_61" class="page bold">Reference</span><br>
                    <ol>
                        <li>
                            Duraj‐Thatte, Anna M, et al. “Hydrogels: Genetically Programmable Self‐Regenerating
                            Bacterial Hydrogels (Adv. Mater. 40/2019).” Advanced Materials, vol. 31, no. 40, 1 Oct.
                            2019, https://doi.org/10.1002/adma.201970289. Accessed 17 Feb. 2024.
                        </li>
                        <li>
                            Duraj‐Thatte, Anna M., et al. “Genetically Programmable Self‐Regenerating Bacterial
                            Hydrogels.” Advanced Materials, vol. 31, no. 40, 12 Aug. 2019, p. 1901826,
                            https://doi.org/10.1002/adma.201901826.
                        </li>
                        <li>
                            Hwang, In Young, et al. “Engineered Probiotic Escherichia Coli Can Eliminate and Prevent
                            Pseudomonas Aeruginosa Gut Infection in Animal Models.” Nature Communications, vol. 8, no.
                            1, Apr. 2017, www.nature.com/articles/ncomms15028, https://doi.org/10.1038/ncomms15028.
                            Accessed 1 Jan. 2020.
                        </li>
                        <li>
                            Mays, Zachary J. S., et al. “Quantifying and Engineering Mucus Adhesion of Probiotics.” ACS
                            Synthetic Biology, vol. 9, no. 2, 7 Jan. 2020, pp. 356–367,
                            https://doi.org/10.1021/acssynbio.9b00356. Accessed 6 Mar. 2022.
                        </li>
                        <li>
                            Praveschotinunt, Pichet, et al. “Engineered E. Coli Nissle 1917 for the Delivery of
                            Matrix-Tethered Therapeutic Domains to the Gut.” Nature Communications, vol. 10, no. 1, Dec.
                            2019, https://doi.org/10.1038/s41467-019-13336-6. Accessed 17 Apr. 2021.
                        </li>
                        <li>
                            Tang, Tzu-Chieh, et al. “Materials Design by Synthetic Biology.” Nature Reviews Materials,
                            vol. 6, no. 4, 23 Dec. 2020, pp. 332–350, https://doi.org/10.1038/s41578-020-00265-w.
                        </li>
                        <li>
                            Tumas, Sarunas, et al. “Engineered E. Coli Nissle 1917 for Delivery of Bioactive IL-2 for
                            Cancer Immunotherapy.” Scientific Reports, vol. 13, no. 1, 2 Aug. 2023, p. 12506,
                            www.nature.com/articles/s41598-023-39365-2, https://doi.org/10.1038/s41598-023-39365-2.
                            Accessed 19 Sept. 2023.
                        </li>
                    </ol>
                </div>
            </div>
        </div>
        <!-- <div class="banner"></div> -->
    </div>
</body>

</html>
<script>
    // let items = document.querySelectorAll(".scroll");
    // let move = document.querySelector(".animation");
    // let last = 0;
    // let h1 = 79.19;
    // let h2 = 64.79;
    // let h3 = 32.4;
    // let h4 = 43.5;

    // for (let i = 0; i < items.length; i++) {
    //     items[i].onclick = function () {
    //         //   items[last].style.backgroundColor = "rgba(126, 185, 187, 0.5)";
    //         // items[last].style.color = "#546ca4"
    //         // this.style.color = "white";
    //         last = i;
    //         if (i <= 0) {
    //             move.style.top = 10 + "px";
    //         }
    //         if (i > 0 && i <= 2) {
    //             move.style.top = 10 + h1 + (i - 1) * h3 + "px";
    //         }
    //         if (i > 2 && i <= 3) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + "px";
    //         }
    //         if (i > 3 && i <= 4) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + "px";
    //         }
    //         if (i > 4 && i <= 5) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + "px";
    //         }
    //         if (i > 5 && i <= 6) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + "px";
    //         }
    //         if (i > 6 && i <= 7) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + "px";
    //         }
    //         if (i > 7 && i <= 8) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + "px";
    //         }
    //         if (i > 8 && i <= 9) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + "px";
    //         }
    //         if (i > 9 && i <= 10) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + "px";
    //         }
    //         if (i > 10 && i <= 11) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + "px";
    //         }
    //         if (i > 11 && i <= 12) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + h1 + "px";
    //         }
    //         if (i > 12) {
    //             move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + h1 + (i - 12) * h3 + "px";
    //         }
    //         move.style.height = 65 + "px";
    //     };
    // }
    // move.addEventListener("transitionend", function () {
    //     move.style.height = "40px";
    // });


    // let lis = document.querySelectorAll('.page');
    // let cover = document.querySelector('.cover');
    // let lastPrintedIndex = null;

    // window.addEventListener('scroll', function () {
    //     let coverRect = cover.getBoundingClientRect();
    //     for (let i = 0; i < lis.length; i++) {
    //         let pageRect = lis[i].getBoundingClientRect();

    //         if (pageRect.bottom >= coverRect.top && pageRect.top <= coverRect.bottom && i !== lastPrintedIndex) {
    //             if (i <= 0) {
    //                 move.style.top = 10 + "px";
    //             }
    //             if (i > 0 && i <= 2) {
    //                 move.style.top = 10 + h1 + (i - 1) * h3 + "px";
    //             }
    //             if (i > 2 && i <= 3) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + "px";
    //             }
    //             if (i > 3 && i <= 4) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + "px";
    //             }
    //             if (i > 4 && i <= 5) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + "px";
    //             }
    //             if (i > 5 && i <= 6) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + "px";
    //             }
    //             if (i > 6 && i <= 7) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + "px";
    //             }
    //             if (i > 7 && i <= 8) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + "px";
    //             }
    //             if (i > 8 && i <= 9) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + "px";
    //             }
    //             if (i > 9 && i <= 10) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + "px";
    //             }
    //             if (i > 10 && i <= 11) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + "px";
    //             }
    //             if (i > 11 && i <= 12) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + h1 + "px";
    //             }
    //             if (i > 12) {
    //                 move.style.top = 10 + h1 + 2 * h3 + 10 + h1 + h3 + h2 + h3 + h2 + 10 + h4 + h3 + h2 + 10 + h1 + (i - 12) * h3 + "px";
    //             }
    //             move.style.height = 65 + "px";
    //             move.addEventListener("transitionend", function () {
    //                 move.style.height = "40px";
    //             });
    //             // items[i].style.color = "white";
    //             // if (i != 0) {
    //             //     items[i - 1].style.color = "#546ca4";
    //             //     items[i + 1].style.color = "#546ca4";
    //             // }
    //             // if (i == 0) {
    //             //     items[i + 1].style.color = "#546ca4";
    //             // }
    //         }
    //     }
    // });

    $(function () {
        $('.caidan').click(function () {
            $('.TOP').toggleClass('TOP-active')
            $('.caidan').toggleClass('caidan-active')
        })
    })

    $("#iconclick").click(function () {
        location.href = 'index.html'
    })

</script>